sqldelight-runtime/src/main/java/com/squareup/sqldelight/ColumnAdapter.kt

```diff
@@ -13,15 +13,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.squareup.sqldelight;
-
-import android.support.annotation.NonNull;
+package com.squareup.sqldelight
 
 /**
- * Marshal and map the type {@code T} to and from a database type {@code S} which is one of
- * {@code Long}, {@code Double}, {@code String}, {@code byte[]}.
+ * Marshal and map the type `T` to and from a database type `S` which is one of
+ * `Long`, `Double`, `String`, `byte[]`.
  */
-public interface ColumnAdapter<T, S> {
-  @NonNull T decode(S databaseValue);
-  S encode(@NonNull T value);
+interface ColumnAdapter<T, S> {
+  fun decode(databaseValue: S): T
+  fun encode(value: T): S
```

Comment:
one interesting consequence of this is that the `T` and the `S` could be declared at the callsite as being nullable

Reply:
does doing `T : Any` prevent that? ie. preventing `T` from being a subtype of `Any?` ?

Reply:
Damn that does work! Cool